controllers/application_controller.go

```diff
@@ -30,12 +31,32 @@ type ApplicationReconciler struct {
 // +kubebuilder:rbac:groups=orkestra.azure.microsoft.com,resources=applications/status,verbs=get;update;patch
 
```

Comment:
Can you add the `updateStatusAndEvent()` as well to the reconcile function

Reply:
done